src/pages/pool/invest.vue

```diff
@@ -37,13 +32,7 @@ useIntervalFn(poolQuery.refetch, oneMinInMs);
 </script>
 
 <template>
-  <JoinPoolProvider
-    v-if="pool && isDeepPool"
-    :pool="pool"
-    :isSingleAssetJoin="activeTab === Tab.SingleToken"
-  >
-    <InvestPage></InvestPage>
-  </JoinPoolProvider>
-  <InvestPage v-else></InvestPage>
+  <!-- TODO: handle loading state -->
```

Comment:
What exactly needs to be handled?

Reply:
Had to move the loading state here, so `provideJoinPool` can be called in the child: `InvestPage.vue`